### PR TITLE
Fixes length for free resolutions

### PR DESCRIPTION
**Pull request by ederc**
Fixes the length of free resolutions computed via `Singular`.

### CODE REVIEW

**Comment by thofma** [comment]
Do we have an example which is fixed by this? If so, we could add it as a test.

**Comment by wdecker** [comment]
The example we discussed earlier would do

S, (x0, x1, x2, x3, x4) = graded_polynomial_ring(GF(3), ["x0", "x1", "x2", "x3", "x4"]);
m = ideal(S, [x1^2+(-x1+x2+x3-x4)*x0, x1*x2+(x1-x3+x4)*x0,
          x1*x3+(-x1+x4+x0)*x0,
          x1*x4+(-x1+x3+x4-x0)*x0, x2^2+(x1-x2-x4-x0)*x0,
          x2*x3+(x1-x2+x3+x4-x0)*x0, x2*x4+(x1+x2-x3-x4-x0)*x0,
          x3^2+(x3+x4-x0)*x0,x3*x4+(-x3-x4+x0)*x0,
          x4^2+(x1+x3-x4-x0)*x0]);
A, _ = quo(S, m);
FA = free_resolution(A)

**Comment by ederc** [comment]
Working on this, give me a moment.

File: src/Modules/UngradedModules/FreeResolutions.jl [hunk 1]
```diff
@@ -204,28 +204,39 @@ function _extend_free_resolution(cc::Hecke.ComplexOfMorphisms, idx::Int)
   dom = domain(cc.maps[1])
   j   = 2
 
-  while j <= Singular.length(res)
+  #= get correct length of Singular sresolution =#
+  slen = iszero(res[Singular.length(res)+1]) ? Singular.length(res) : Singular.length(res)+1
```
Comment (HechtiDerLachs) [inline src/Modules/UngradedModules/FreeResolutions.jl:208]: Unfortunately, this has not worked for me some days ago. I tried different examples and a fixed offset did not work out. Are you sure this does the job beyond the example provided by @wdecker ?

Comment (ederc) [reply src/Modules/UngradedModules/FreeResolutions.jl:208]: According to @hannes14 this should be the way to go, I tried several examples and it worked. If you have some counterexample just let me know.

**Review by hannes14 — APPROVED** [review]

File: src/Modules/UngradedModules/FreeResolutions.jl (hunk 1 above)
Comment (HechtiDerLachs) [reply src/Modules/UngradedModules/FreeResolutions.jl:208]: Sorry, I didn't read it correctly. Probably this works.

**Review by wdecker — APPROVED** [review]

**Pull request merged**